publish2cloud.py

```diff
@@ -176,8 +176,19 @@ def publish_to_s3(config, section, chunknum):
     if config.has_option(section, 's3_key'):
         key = config.get(section, 's3_key')
 
-    chunk_key = os.path.join(
-        config.get(section, os.path.basename('output')), str(chunknum))
+    versioning_needed = (
+        config.has_option(section, 'versioning_needed')
+        and config.getboolean(section, 'versioning_needed')
+    )
+    if versioning_needed:
+        chunk_key = os.path.join(
+            config.get(section, os.path.basename('output')),
+            config.get(section, 'chunk_key_version'),
+            str(chunknum)
+        )
+    else:
+        chunk_key = os.path.join(
+            config.get(section, os.path.basename('output')), str(chunknum))
```

Comment:
```suggestion
            config.get(section, os.path.basename('output')),
            str(chunknum)
        )
```

Reply:
I tripped up trying to compare this to the above logic ... this change made it more read-able and scan-able for me.